internal/controller/clusterdeployment_controller.go

```diff
@@ -271,6 +274,13 @@ func (r *ClusterDeploymentReconciler) updateCluster(ctx context.Context, cd *kcm
 			"namespace":  cred.Spec.IdentityRef.Namespace,
 		}
 
+		if r.GlobalRegistry != "" || r.GlobalK0sURL != "" {
+			values["global"] = map[string]any{
+				"registry": r.GlobalRegistry,
```

Comment:
I think still, extra conditions are required here: set `registry` only if `r.GlobalRegistry` is not empty, and set `k0sURL` only if `r.GlobalK0sURL` is not empty. But I guess these parameters work in a bundle only, right?

Reply:
I assume that they will be passed together into `ClusterDeployment` config. But I guess it won't hurt to add an extra check